### ISSUE TITLE
Adding waypoints

### ISSUE THREAD

**Issue by Hard-Loader** [Project-OSRM/osrm-frontend-v2#343]
With a freshly loaded osrm-frontend, clicking on the map sets the start marker and a second click sets the end marker.

Subsequent clicks always update the second waypoint, whether or not extra waypoint spaces have been added or selected.  This does not seem like expected behaviour. I would expect each subsequent click to add a marker after the currently selected list entry.

Is this something that might be changed in future?

I am currently testing a script that inserts routes into the map by repeatedly calling the addWaypoint() function.  This only works for the start and end points, for the reasons above. Is there a workaround that would allow each point to be assigned in order to the list?

**Comment by Hard-Loader** [comment]
I've taken a closer look at what's going on.  It's an easy change to make new markers add to the list rather than over-writing the last value.  Line 204 in /src/index.js currently reads as "if (length === 1) length = length + 1;"  changing that to "if (length > 0) length = length + 1;" will add a new entry for each click on the map.  

I think that's more intuitive behaviour than the way it works at the moment.

**Comment by mjjbell** [comment]
> Subsequent clicks always update the second waypoint, whether or not extra waypoint spaces have been added or selected. This does not seem like expected behaviour. I would expect each subsequent click to add a marker after the currently selected list entry.

I agree, this does feel unintuitive. Your suggestion makes sense to me.

**Comment by robinsonkwame** [comment]
Note: if you copy and paste another address and enter it into an empty via point and press enter you can add more than a stop and start. Very odd but it is what it is.

**Comment by datendelphin** [comment]
I think the main intended "work flow" for via points is dragging the route. But it makes sense to just add waypoints with each click.

**Comment by DennisOSRM** [comment]
This has been changed now: First two clicks set origin and destination as expected. Subsequent clicks don't change the location of any marker. To delete you can click on the respective marker, or move it with drag and drop. Setting via points can be done by CTRL-click (or command-click on macOS). Via points are added to the route in the order they are placed on the map. Click-to-remove and drag-and-drop work the same.

This should clear up the confusing UX. Let me know if you have any questions.
